src/transformers/models/jamba/modeling_jamba.py

```diff
@@ -0,0 +1,2127 @@
+                self.out_proj.weight,
+                self.out_proj.bias.float() if self.use_bias else None,
+                -torch.exp(self.A_log.float()),
+                None,  # input-dependent B
+                None,  # input-dependent C
+                self.D.float(),
+                delta_bias=self.dt_proj.bias.float(),
+                delta_softplus=True,
+            )
+
+        else:
+            hidden_states, gate = projected_states.chunk(2, dim=1)
+
+            # 2. Convolution sequence transformation
+            conv_weights = self.conv1d.weight.view(self.conv1d.weight.size(0), self.conv1d.weight.size(2))
+            if cache_params is not None and cache_params.seqlen_offset > 0:
+                hidden_states = causal_conv1d_update(
+                    hidden_states.squeeze(-1),
+                    cache_params.conv_states[self.layer_idx],
+                    conv_weights,
+                    self.conv1d.bias,
+                    self.activation,
+                )
+                hidden_states = hidden_states.unsqueeze(-1)
+            else:
+                if cache_params is not None:
+                    conv_states = nn.functional.pad(
+                        hidden_states, (self.conv_kernel_size - hidden_states.shape[-1], 0)
+                    )
+                    cache_params.conv_states[self.layer_idx].copy_(conv_states)
+                hidden_states = causal_conv1d_fn(
+                    hidden_states, conv_weights, self.conv1d.bias, activation=self.activation
+                )
+
+            # 3. State Space Model sequence transformation
+            # 3.a. input varying initialization of time_step, B and C
+            ssm_parameters = self.x_proj(hidden_states.transpose(1, 2))
+            time_step, B, C = torch.split(
+                ssm_parameters, [self.time_step_rank, self.ssm_state_size, self.ssm_state_size], dim=-1
+            )
+            time_step, B, C = self._apply_layernorms(time_step, B, C)
+
+            # Here we need to apply dt_proj without the bias, as the bias is added in the selective scan kernel.
+            # This is a hack to apply dt_proj while still using the forward pass of `torch.nn.Linear`, which is needed
+            # in order to make quantization work. Quantization code replaces `torch.nn.Linear` layers with quantized
+            # linear layers, and requires to call the forward pass directly.
+            # The original code here was: ```discrete_time_step = self.dt_proj.weight @ time_step.transpose(1, 2)```
+            if hasattr(self.dt_proj, "base_layer"):
+                # In case of LoRA, we need to access the base layer to get the bias
+                time_proj_bias = self.dt_proj.base_layer.bias
+                self.dt_proj.base_layer.bias = None
+            else:
+                time_proj_bias = self.dt_proj.bias
+                self.dt_proj.bias = None
+            discrete_time_step = self.dt_proj(time_step).transpose(1, 2)
```

Comment:
https://github.com/huggingface/peft/pull/1530 should be able to clear this no?

Reply:
Yes. removed